File: Children-Competition-Java/mpp-proiect-java-ABNegrea/Problema5/src/main/java/Concurs/Repository/Interfaces/Repository.java
package Concurs.Repository.Interfaces;
import Concurs.Domain.Entity;


import java.util.Collection;

public interface Repository<ID, E extends Entity<ID>> {
    E findByID(ID id) throws IllegalArgumentException;

    Collection<E> findAll();

    E save(E entity) throws IllegalArgumentException;

    E remove(ID id) throws IllegalArgumentException;

    E update(ID id, E entity) throws IllegalArgumentException;
}

File: School-Tournament/Proiect_Bonus/Repository/Repository.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Proiect_Bonus.Domain;

namespace Proiect_Bonus.Repository
{
    internal interface Repository<ID, E> where E: Entity<ID>
    {
        E findOne(ID id);

        List<E> findAll();

        E save(E entity);

        E delete(ID id);

        E update(E entity, E newEntity);
    }
}
